graphql/lecture/fields.ts

```diff
@@ -0,0 +1,16 @@
+import { Role } from '../authorizations';
+import { Arg, Authorized, Ctx, FieldResolver, Query, Resolver, Root } from 'type-graphql';
+import { Course, Lecture, Subcourse, Pupil, Bbb_meeting as BBBMeeting } from '../generated';
+import { GraphQLContext } from '../context';
+import { getSessionStudent } from '../authentication';
+import { prisma } from '../../common/prisma';
+
+@Resolver((of) => Lecture)
+export class ExtendedFieldsLectureResolver {
+    @FieldResolver((returns) => Boolean)
+    @Authorized(Role.ADMIN, Role.STUDENT)
+    async isInstructor(@Ctx() context: GraphQLContext, @Root() subcourse: Subcourse, @Arg('studentId', { nullable: true }) studentId: number) {
+        const student = await getSessionStudent(context, studentId);
+        return (await prisma.lecture.count({ where: { instructorId: student.id } })) > 0;
```

Comment:
Missing lecture id, checking lecture.instructorId = student.id should be enough (and the resolver is slightly useless then, although maybe good for consistency, haven't thought this through...)

Reply:
Also Root is not a Subcourse?